concierge/service/src/main/java/org/eclipse/ditto/concierge/service/enforcement/CreationRestrictionEnforcer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.concierge.service.enforcement;
+
+import org.eclipse.ditto.base.model.auth.AuthorizationContext;
+
+/**
+ * Interface for evaluating if the entity creation should be restricted.
+ */
+@FunctionalInterface
+public interface CreationRestrictionEnforcer {
+
+    /**
+     * The context for evaluating if an entity may be created or not.
+     */
+     class Context {
+        private final String resourceType;
+        private final String namespace;
+        private final AuthorizationContext authorizationContext;
```

Comment:
By adding the complete `DittoHeaders` instead of only the `authorizationContext` to the `Context`, we could also use the `correlationId` in the added log statements (in `DefaultCreationRestrictionEnforcer`) - that way requests are far easier traceable in the logs.